src/components/MainNav/CurrentApp/CurrentApp.js

```diff
@@ -4,22 +4,45 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
-import NavButton from '../NavButton';
+import AppIcon from '@folio/stripes-components/lib/AppIcon';
+import Badge from '@folio/stripes-components/lib/Badge';
+import Headline from '@folio/stripes-components/lib/Headline';
+import css from './CurrentApp.css';
 
 const propTypes = {
-  currentApp: PropTypes.object,
+  currentApp: PropTypes.shape(
+    {
+      displayName: PropTypes.string,
+      description: PropTypes.string,
+    },
+  ),
+  id: PropTypes.string,
+  icon: PropTypes.oneOfType([
+    PropTypes.element,
+  ]),
+  badge: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
 };
 
-const CurrentApp = ({ currentApp }) => {
-  const { displayName, description } = currentApp;
+const defaultProps = {
+  currentApp: { displayName: 'FOLIO', description: 'FOLIO platform' },
+};
+
+const CurrentApp = ({ currentApp, icon, badge, id }) => {
+  const displayIcon = (<div className={css.icon}>{icon || <AppIcon focusable={false} />}</div>);
+
   return (
-    <NavButton
-      label={displayName}
-      title={description}
-    />
+    <div id={id} title={currentApp.description} className={css.currentApp}>
+      {badge && (<Badge color="red" className={css.badge}>{badge}</Badge>)}
+      {displayIcon}
+      <Headline tag="h1" size="small" style={{ margin: 0 }} >{currentApp.displayName}</Headline>
```

Comment:
There is a margin prop for the Headline-component that can be set to "none" instead of using inline styles. I just realized that the option isn't documented in the readme of the Headline-component so I'll update that in stripes-components.

Reply:
Alright.. will update.